### ISSUE TITLE
Feature Request: Timeline view?

### ISSUE THREAD

**Issue by csharpen** [aardappel/procrastitracker#15]
First of all - thanks for all the great work on this incredible app! It's amazingly simple and useful.

Have you considered also showing a timeline view, in addition to the tree? When reviewing the work I've done each day, it's incredibly helpful to piece the day together with window activities. I assume this would deviate slightly, as it's likely that the same window would appear multiple times throughout the day, so would need to be tracked as such. There are commercial products doing this, but it'd be great if procrastitracker could fit the bill.

**Comment by aardappel** [comment]
The problem is that that it doesn't actually keep this information, i.e. it knows you spent X hrs in app Y during day Z, but it doesn't actually know when during the day you spent that time.

This was a design decision from the start to keep the database compact (which is somewhat useful in hindsight, given that my personal database now spans 9 years, and is still only 1MB in size :)

It does have the start time for each type of time use during the day, so some sort of approximation is possible, but it wouldn't look like an accurate display of your day.

**Comment by csharpen** [comment]
I appreciate that it's been kept to an efficient format. Concern for resources is a pretty rare trait in applications these days. However, in this case I think it'd be really useful info to handy - perhaps there could be a flag that would extend the current DB, or add tracking to a second DB optionally? It'd be incredibly useful to anyone who needs to account for their schedule for administrative, billing, etc purposes.

Thanks again for your time developing on the app!

**Comment by aardappel** [comment]
I agree that this would be nice to have, and it could be done (it could store multiple day segments for a single day when data is non-contiguous), but certainly isn't trivial as this data change would affect almost all code.

Also, writing the actual UI to display it is very clumsy (again, for efficiency, it is all written in plain WIN32, which is painful).

**Comment by gjcoll** [comment]
I was curious if this is still the case from 2016. I am specifically interested in monitoring the start and stop times of idle time on my computer.

**Comment by aardappel** [comment]
@gjcoll Yup, the data format hasn't changed.

**Comment by vinceswann** [comment]
I too was looking for this feature. I hope it gets added in sometime in future.

**Comment by aardappel** [comment]
@vinceswann noted

**Comment by sulisu** [comment]
I am happy to know someone else have the same requirement as me.

Not directly related to procrastitracker, but these two AHK scripts meet my requirement to some extent.

https://www.autohotkey.com/board/topic/57681-how-to-modify-a-computer-usage-tracker-to-track-idle-time/
https://www.autohotkey.com/board/topic/37998-tiny-computer-usage-tracker/
